pathetic-model/src/main/java/org/patheloper/model/pathing/pathfinder/AStarPathfinder.java

```diff
@@ -190,26 +223,40 @@ private boolean isNodeInvalid(Node node, List<PathFilter> filters) {
       }
     }
 
-    return !isWithinWorldBounds(node.getPosition()) || !doAllFiltersPass(filters, node);
+    return !isWithinWorldBounds(node.getPosition())
+        || !doAllFiltersPass(filters, node) && !doAnyFilterStagePass(filterStages, node);
   }
 
   private boolean doAllFiltersPass(List<PathFilter> filters, Node node) {
-    Map<Class<? extends PathFilter>, Boolean> cache = new HashMap<>();
-
     for (PathFilter filter : filters) {
       PathValidationContext context =
           new PathValidationContext(
               node.getPosition(),
               node.getParent() != null ? node.getParent().getPosition() : null,
               snapshotManager);
 
-      if (!cache.computeIfAbsent(filter.getClass(), k -> filter.filter(context))) {
+      if (!filter.filter(context)) {
         return false;
       }
     }
     return true;
   }
 
+  private boolean doAnyFilterStagePass(List<PathFilterStage> filterStages, Node node) {
+    if (filterStages.isEmpty()) return true;
+
+    for (PathFilterStage filterStage : filterStages) {
+      if (filterStage.filter(
+          new PathValidationContext(
+              node.getPosition(),
+              node.getParent() != null ? node.getParent().getPosition() : null,
+              snapshotManager))) {
+        return true;
+      }
+    }
+    return pathfinderConfiguration.isPrioritizing();
```

Comment:
In case that prioritizing is active, that the filterstages wont be count into the validation process. We could probably move that upwards, one scope higher.